### PR TITLE
Split performances & scalability guides

### PR DESCRIPTION
**Pull request by nidup**

### CODE REVIEW

File: reference/performances_guide/mongodb_product_saving.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+What about MongoDB storage and bulk product saving performances?
+----------------------------------------------------------------
+
+In the version 1.3, we had a default product saver which handled the save of a single product or a set of products in the database.
```
Comment (grena) [inline reference/performances_guide/mongodb_product_saving.rst]: >In the 1.3 version?

File: reference/scalability_guide/more_than_10k_families.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    If you encounter one of these problems, don't hesitate to take regularly a look at our `changelog`_. If you encounter another problem, please contact us.
+
+.. _changelog: https://github.com/akeneo/pim-community-dev/blob/1.4/CHANGELOG-1.4.md
+
+.. warning::
+
+    On this axis, we still have different issues. We've tested with 10k families and a low amount of attributes and categories.
+
+    Screens impacted are the following and we're releasing performances fixes as 1.4 patches:
+      - **[fixed in 1.4.12]** (PIM-5194) edition of a product and change its family (Community Edition)
```
Comment (willy-ahva) [inline reference/scalability_guide/more_than_10k_families.rst]: Given users have no access to Jira, maybe add PIM-xxxx is useless in the doc.

File: reference/performances_guide/mongodb_product_saving.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+This default saver was the same for Doctrine ORM and Doctrine MongoDB and it relied on Doctrine Common ObjectManager interface, it uses the default persist/flush operations.
+
+These operations were especially greedy. In the 1.3, to prevent performance issues, we provided another method for the product import :doc:`/cookbook/import_export/mongodb-fast-writer`
+
+Mainly this new method bypass Doctrine MongoDB by normalizing the product objects to final Mongodb Documents to be able to directly save them in the database. It still uses low-level Doctrine methods. It simply doesn't use the UnitOfWork.
```
Comment (grena) [inline reference/performances_guide/mongodb_product_saving.rst]: > Mainly this new method **bypasses** Doctrine MongoDB by normalizing the product objects to final Mongodb Documents **in order to** be able to directly save them in the database.

File: reference/scalability_guide/representative_catalogs.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
+Audit with 3 Representative Catalogs
+====================================
+
+We've audited the application with 3 different catalogs, the data sets are available here https://github.com/akeneo/catalogs/tree/1.4
```
Comment (willy-ahva) [inline reference/scalability_guide/representative_catalogs.rst]: 3 different catalogs, data sets are available

File: reference/scalability_guide/more_than_100k_products_to_export.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
+More than 100k products to export?
+----------------------------------
+
+We had use cases where customers would export 270k products and had issues with the memory usage.
+
+Most of the massive operations of the PIM like imports and exports are bulked by page of lines (or objects) to avoid too large memory usage.
+
+As each product can have different properties, export kept the transformed array in memory to add missing columns from a line to another.
+
+In the version 1.4.9 (PIM-5127), we changed the internal behavior of the CsvProductWriter to use a file buffer to temporary write each transformed array before to aggregating the final result.
```
Comment (grena) [inline reference/scalability_guide/more_than_100k_products_to_export.rst]: > (...) a file buffer to **temporarily** write each **previously transformed** array **in order to aggregate** the final result.

File: reference/scalability_guide/more_than_100k_products_to_export.rst [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+As a conclusion, for product export, limitation is no more available memory but available hard drive space.
```
Comment (grena) [inline reference/scalability_guide/more_than_100k_products_to_export.rst]: > As a conclusion, for product export, limitation is **now the hard drive space and no longer the available memory.**

File: reference/scalability_guide/more_than_100k_products_to_export.rst [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+Please notice that the number of values per product will have an impact on the time of execution and memory usage.
```
Comment (grena) [inline reference/scalability_guide/more_than_100k_products_to_export.rst]: > (...) on the **execution time** and memory usage.

File: reference/scalability_guide/more_than_64_indexes_with_mongodb.rst [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    + ( N localized attributes usable as filters * N enabled locales )
+    + ( N scopable attributes usable as filters * N existing channels )
+    + ( N scopable AND localizable attributes usable as filters * N enabled locales * N existing channels )
+    + N enabled locales * N enabled channels (for the completeness filters)
+    + 3 for family, groups and categories
+    > 64
+
+.. warning::
+
+    If your collection of products requires more than 64 indexes, please contact us (we've developed a Elastic Search implementation to get rid of this limit).
```
Comment (grena) [inline reference/scalability_guide/more_than_64_indexes_with_mongodb.rst]: > (...) developed **an** Elastic Search (...)

**Comment by grena** [comment]
Commented then GTM!

File: reference/performances_guide/mongodb_product_saving.rst (hunk 1 above)
Comment (nidup) [inline reference/performances_guide/mongodb_product_saving.rst]: yes

File: reference/scalability_guide/more_than_10k_families.rst (hunk 2 above)
Comment (nidup) [inline reference/scalability_guide/more_than_10k_families.rst]: useful for us, to communicate with EE customer and to prepare release notes

**Pull request merged**